core/lib/Cardano/Address/Style/Icarus.hs

```diff
@@ -311,16 +315,19 @@ deriveAddressPublicKey =
 -- > "addr1vxpfffuj3zkp5g7ct6h4va89caxx9ayq2gvkyfvww48sdncxsce5t"
 --
 -- | Analyze an 'Address' to know whether it's an Icarus address or not.
--- Returns 'Nothing' if the address isn't a byron address, or return a
+-- Throws 'IcarusAddrError' if the address isn't a byron address, or return a
 -- structured JSON that gives information about an address.
 --
 -- @since 2.0.0
-inspectIcarusAddress :: Address -> Maybe Json.Value
+inspectIcarusAddress :: MonadThrow m => Address -> m Json.Value
 inspectIcarusAddress addr = do
-    payload <- CBOR.deserialiseCbor CBOR.decodeAddressPayload bytes
-    (root, attrs) <- CBOR.deserialiseCbor decodePayload payload
-    guard $ 1 `notElem` (fst <$> attrs)
-    ntwrk <- CBOR.deserialiseCbor CBOR.decodeProtocolMagicAttr payload
+    payload <- either (throwM . IcDeserialiseError) pure
+      $ CBOR.deserialiseCbor CBOR.decodeAddressPayload bytes
+    (root, attrs) <- either (throwM . IcDeserialiseError) pure
+      $ CBOR.deserialiseCbor decodePayload payload
+    when (elem 1 . fmap fst $ attrs) $ throwM IcUnexpectedDerivationPath
+    ntwrk <- either (throwM . IcDeserialiseError) pure
+      $ CBOR.deserialiseCbor CBOR.decodeProtocolMagicAttr payload
```

Comment:
side-note, your editor should be configured to pick up the configuration at the root of the repository:  https://github.com/input-output-hk/cardano-addresses/blob/master/.editorconfig

This specifies, among other things, that we use 4-space indentation for Haskell code. Make sure your editor is correctly configured :+1: